ebms-async/src/main/kotlin/no/nav/emottak/ebms/async/kafka/consumer/FailedMessageKafkaHandler.kt

```diff
@@ -0,0 +1,173 @@
+    val kafkaErrorQueue: KafkaErrorQueue = config().kafkaErrorQueue,
+    kafka: Kafka = config().kafka
+) {
+
+    private var producer = KafkaProducer(
+        kafka.toProperties(),
+        StringSerializer(),
+        ByteArraySerializer()
+    )
+
+    private var producersFlow: Flow<KafkaProducer<String, ByteArray>> = kafkaProducer( // TODO Deprecated
+        ProducerSettings(
+            bootstrapServers = kafka.bootstrapServers,
+            keyDeserializer = StringSerializer(),
+            valueDeserializer = ByteArraySerializer(),
+            acks = Acks.All,
+            other = kafka.toProperties()
+        )
+    )
+    private var consumerFlow: Flow<ReceiverRecord<String, ByteArray>> = KafkaReceiver(
+        ReceiverSettings(
+            bootstrapServers = kafka.bootstrapServers,
+            keyDeserializer = StringDeserializer(),
+            valueDeserializer = ByteArrayDeserializer(),
+            groupId = kafka.groupId,
+            pollTimeout = 10.seconds,
+            properties = kafka.toProperties()
+        )
+    ).receive(kafkaErrorQueue.topic)
+
+    suspend fun send(record: ReceiverRecord<String, ByteArray>, key: String = record.key(), value: ByteArray = record.value()) {
+        record.addHeader(RETRY_AFTER, getNextRetryTime(record))
+        try {
+            val result = producer.send(ProducerRecord(kafkaErrorQueue.topic, null, key, value, record.headers())).get()
+            logger.info("Wrote to offset:" + result.offset())
+            producer.close()
+//            producersFlow.collect { producer ->
+//                val metadata = producer.send(ProducerRecord(kafkaErrorQueue.topic, null, key, value, record.headers())).get()
+//                logger.info("Offset on metadata: " + metadata.offset())
+//                logger.info("Result " + metadata.partition() + " timestamp " + metadata.timestamp())
+//                producer.commitTransaction()
+//                producer.close()
+//            }
+            logger.info("Message sent successfully to topic ${kafkaErrorQueue.topic}")
+        } catch (e: Exception) {
+            logger.info("Failed to send message to ${kafkaErrorQueue.topic} : ${e.message}")
+        }
+    }
+
+    suspend fun receive(payloadMessageProcessor: PayloadMessageProcessor, limit: Int = 10) { // TODO limit til offset
+        logger.debug("Reading from error queue")
+        var counter = 0
+        consumerFlow.map { record ->
+            counter++
+            if (counter > limit) {
+                throw Exception("Error queue limit exceeded: $limit") // TODO fjern dette
+            }
+            record.offset.acknowledge()
+            record.retryCounter()
+            payloadMessageProcessor.process(record)
+            record.offset.acknowledge()
+        }.collect()
+    }
+
+    fun getNextRetryTime(record: ReceiverRecord<String, ByteArray>): String {
+        return DateTime.now().plusMinutes(5)
+            .toString() // TODO create retry strategy
```

Comment:
Ref retry strategi, kanskje du bør se litt på https://arrow-kt.io/learn/resilience/retry-and-repeat/